bfvmm/include/serial/serial_port_ns16550a.h

```diff
@@ -16,14 +16,19 @@
 // License along with this library; if not, write to the Free Software
 // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
-#ifndef SERIAL_PORT_INTEL_X64_H
-#define SERIAL_PORT_INTEL_X64_H
+#ifndef SERIAL_PORT_NS16550A_H
+#define SERIAL_PORT_NS16550A_H
 
 #include <string>
 #include <memory>
 
 #include <bfconstants.h>
-#include <intrinsics/x86/common_x64.h>
+
+#if defined(__x86_64__)
```

Comment:
I'd love this logic to be in a header. Something like just including common.h, and it figures out which files you need. Otherwise, a lot of files are going to have this same logic, which will be annoying to fix later when we add AMD support

Reply:
This is what I mentioned in chat we'll have to discuss. `__x86_64__` isn't defined on MSVC. We need to decide how we're going to handle that (part of which is me reading the existing code to see if there's already something - there may be, I'll RFC when I'm more informed)